src/pdf_parser/pdf_utils/ocr.py

```diff
@@ -0,0 +1,459 @@
+import concurrent.futures
+import io
+from collections import defaultdict
+from typing import Optional, Tuple, List, Union
+
+import numpy as np
+from PIL.PpmImagePlugin import PpmImageFile
+from google.cloud import vision
+from google.cloud.vision import types
+from google.cloud.vision_v1.types import BoundingPoly
+from layoutparser import TextBlock, Rectangle, Layout
+from layoutparser.ocr import TesseractAgent, GCVAgent
+from pydantic import BaseModel as PydanticBaseModel
+from shapely.geometry import Polygon
+
+from src.base import PDFTextBlock
+from src.pdf_parser.pdf_utils.disambiguate_layout import lp_coords_to_shapely_polygon
+
+
+class BaseModel(PydanticBaseModel):
+    """Base class for all models."""
+
+    class Config:
+        """Pydantic config."""
+
+        arbitrary_types_allowed = True
+
+
```

Comment:
Possibly some duplication here with the PDFTextBlock models, but created these classes for specificity.